src/components/matrix/column-tree-headers.tsx

```diff
@@ -0,0 +1,592 @@
+import SharedColumnHeaders, { SharedColumnHeadersProps } from '@isrd-isi-edu/deriva-webapps/src/components/matrix//shared-column-headers';
+
+// Data type used for treeview
+import { ParsedGridCell } from '@isrd-isi-edu/deriva-webapps/src/hooks/matrix';
+import { MatrixTreeDatum } from '@isrd-isi-edu/deriva-webapps/src/hooks/matrix';
+import { TreeNode } from '@isrd-isi-edu/deriva-webapps/src/hooks/matrix';
+import { TreeNodeMap } from '@isrd-isi-edu/deriva-webapps/src/hooks/matrix';
+
+// MUI tree components
+import TreeView from '@mui/lab/TreeView';
+import { alpha, styled } from '@mui/material/styles';
+import TreeItem, { TreeItemProps, treeItemClasses, useTreeItem, TreeItemContentProps } from '@mui/lab/TreeItem';
+import clsx from 'clsx';
+import Typography from '@mui/material/Typography';
+
+
+type ColumnHeadersProps = SharedColumnHeadersProps & {
+  /**
+   * x hierarchical data passed to each column
+   */
+  treeData: MatrixTreeDatum[];
+  /**
+   * x hierarchical data nodes passed to each column
+   */
+  treeNodes: TreeNode[];
+  /**
+   * x hierarchical data nodes map passed to each column
+   */
+  treeNodesMap: TreeNodeMap;
+  /**
+   * on scroll event
+   */
+  onScroll?: any;
+};
+
+/**
+ * Virtualized Column Header that displays headers as they scroll into the given width
+ */
+const ColumnHeaders = (props: ColumnHeadersProps, ref: ForwardedRef<any>): JSX.Element => {
+
+  const {
+    searchedColID,
+    listData,
+    setFilteredGridXData,
+    setVisiableColNodes,
+    scrollTreeYIniPos,
+    isScrolling } = props.itemData;
+
+  const [prevSearched, setPrevSearched] = useState<string | null>(null); // previous searched enrty
+  const [expanded, setExpanded] = useState<string[]>([]); // all expanded nodes
+  const [treeDataDict, setTreeDataDict] = useState<Record<string, MatrixTreeDatum>>({}); // Dictionary to store relationship of parent and child for tree data
+  
+  const [scrollableHeight, setScrollableHeight] = useState<number>(props.scrollableMaxHeight);
+  const divRef = useRef<any>(null);
+  /**
+   * styles for tree view and tree items
+   */
+  // the font size is 14 for this page
+  const iconSize = 14;
+
+  // style for the whole tree
+  const columnTreeHeadersStyles: CSSProperties = {
+    overflowX: 'auto',
+    overflowY: 'hidden',
+    height: scrollableHeight,
+    willChange: 'transform',
+  };
+
+  /**
+   * Dynamically adjust the width of scrollable content
+   */
+  useEffect(() => {
+    if (props.scrollable && props.scrollableMaxHeight === -1){
+      // Access the current width here
+      const currentHeight = divRef.current.clientWidth;
+      if (currentHeight !== undefined) {
+        setScrollableHeight(currentHeight);
+      }
+    }
+  }, [expanded]);
+
+  /**
+   * Check whether all ancestors of a node exist in visitedNode list
+   */
+  const checkParentChainExist = (treeDataDict: Record<string, MatrixTreeDatum>, nodeId: string, visitedNodes: Set<string>): boolean => {
```

Comment:
These two functions are identical between `column-tree-headers.tsx` and `row-tree-headers.tsx`. Create a `tree.ts` under `src/utils` and move these functions in there.